Add back setting locale for gettext

janvorli opened the pull request:
Add back setting locale for gettext that was accidentally removed by change #736. I have just made it friendly to the app using coreclr / PAL so that it sets the LC_MESSAGE locale category only for the gettext API call and reverts it back after the call.

janvorli commented:
@jkotas Can you take a look please?

janvorli commented:
Hmm, I've thought it is per thread but it is not. So I need to do it the way it was done before, in the PAL_InitializeCoreCLR for now.

janvorli commented:
@jkotas I've modified it to what it was before the locale setting got accidentally nuked.

jkotas commented on src/pal/src/locale/unicode.cpp:
```diff
@@ -159,6 +159,22 @@ BOOL GetUnicodeData(INT nUnicodeValue, UnicodeDataRec *pDataRec)
+ -    the native resources string
+ ---*/
+void CODEPAGEInit( void )
+{
```
This method has nothing to do with codepages now. Maybe it would be best to call this in PAL_BindResources? (Or rename the method and move it next to PAL_BindResources.)

janvorli commented:
I've renamed the method and also removed the `#ifdefs` and moved it into the method itself since we might need some initialization for the OSX as well once we add the resource string support there.

jkotas commented:
LGTM
